test/web-platform-tests/index.js

```diff
@@ -301,6 +301,11 @@ describe("Web Platform Tests", () => {
     "html/syntax/parsing/template/clearing-the-stack-back-to-a-given-context/clearing-stack-back-to-a-table-context.html",
     "html/syntax/parsing/template/clearing-the-stack-back-to-a-given-context/clearing-stack-back-to-a-table-row-context.html",
     // "html/syntax/parsing/template/creating-an-element-for-the-token/template-owner-document.html", // flaky on CI; possibly due to how we load iframes
+    "html/webappapis/animation-frames/callback-exception.html",
+    // "html/webappapis/animation-frames/callback-invoked.html", document.hidden should be false, but is true
```

Comment:
This test points to a real problem, which is that we now have a contradiction within jsdom. jsdom signals itself as a "headless" browser, which does not display things visually. Thus hidden is true. But for such browsers, the spec says that requestAnimationFrame should not be called. So I am not sure what to do here.

Reply:
IMO the spec is being overly strict, but as I have no real experience with specs and all the edge cases they have to cover, I'll of course defer to you.

But I would like to avoid having to polyfill this for projects with tests run in a browser environment 🙂

Reply:
What I as a developer find more natural is "a headless browser should have it's callback invoked immediately, not at what matches 60 FPS (or the browser's refreshrate)" or something along those lines.

Reply:
I mean, no animations can run in a headless browser, so I think the spec makes sense.

Reply:
My current thinking is maybe we need a new top-level jsdom option, like `pretendToBeVisual: true`, which opts you out of document.hidden and opts you in to requestAnimationFrame. Curious what other contributors think.

(And also better names. Although it needs to convey that this is "pretend" or "fake" or something, since we *don't* run e.g. CSS animations or do layout or paint or any other visual stuff.)

Reply:
I like that idea.
Does it have to be done in this PR, or is this blocked on a new option?

Reply:
I think we need such an option before we can add requestAnimationFrame. But, you probably shouldn't spend much time on it before we give the other contributors and team members a chance to wake up and weigh in. Maybe they'll have a brilliant alternate suggestion we should do instead.

Reply:
I'm positive towards the idea of an option, as long as the README makes the limitations of this mode clear. It should probably also affect `document.visibilityState`.

In the longer term this could potentially be used to resolve issues like #1422, with `window.scroll()` logging to the console by default but pretending to scroll a viewport when this option is enabled. We may want to make the value of the option an object or a string (akin to `runScripts`) to enable different "levels" of pretending, in this case.

Reply:
We're hitting this more often nowadays when implementing new APIs, I do think it's time for a new config option...